x/curating/keeper/post.go

```diff
@@ -35,35 +35,47 @@ func (k Keeper) GetPost(
 
 // CreatePost registers a post on-chain and starts the curation period.
 // It can be called from CreateUpvote() when a post doesn't exist yet.
-func (k Keeper) CreatePost(ctx sdk.Context, vendorID uint32, postID types.PostID,
-	bodyHash types.BodyHash, body string, creator, rewardAccount sdk.AccAddress) error {
-
-	_, found, err := k.GetPost(ctx, vendorID, postID)
-	if err != nil {
-		return err
-	}
-	if found {
-		return types.ErrDuplicatePost
-	}
+func (k Keeper) CreatePost(ctx sdk.Context, vendorID uint32, postID *types.PostID,
+	bodyHash types.BodyHash, body string, creator, rewardAccount sdk.AccAddress) (post types.Post, err error) {
 
 	err = k.validateVendorID(ctx, vendorID)
 	if err != nil {
-		return err
+		return post, err
 	}
 	err = k.validatePostBodyLength(ctx, body)
 	if err != nil {
-		return err
+		return post, err
 	}
 	if rewardAccount.Empty() {
 		rewardAccount = creator
 	}
 
 	curationWindow := k.GetParams(ctx).CurationWindow
 	curationEndTime := ctx.BlockTime().Add(curationWindow)
-	post := types.NewPost(vendorID, postID, bodyHash, body, creator, rewardAccount, curationEndTime)
+
+	if vendorID != 0 && postID != nil {
```

Comment:
I guess we should only check `postID!=nil`  otherwise native posts won't be checked for duplicate?